cardano-testnet/src/Testnet/Runtime.hs

```diff
@@ -228,18 +233,46 @@ startNode tp@(TmpAbsolutePath _tempAbsPath) node port nodeCmd = GHC.withFrozenCa
              , IO.cwd = Just tempBaseAbsPath
              }
 
- -- H.noteShowM_ $ H.getPid hProcess
+  -- We force the evaluation of initiateProcess so we can be sure that
+  -- the process has started. This allows us to read stderr in order
+  -- to fail early on errors generated from the cardano-node binary.
+  mpid <- liftIO $ IO.getPid hProcess
+
+  when (isNothing mpid)
+    $ left $ NodeExecutableError $ mconcat ["startNode: ", node, "'s process did not start."]
+
+  -- The process should have started so we wait a short amount of time to allow
+  -- stderr to be populated with any errors from the cardano-node binary.
+  liftIO $ IO.threadDelay 5_000_000
+
+  stdErrContents <- liftIO $ IO.readFile nodeStderrFile
+
+  if null stdErrContents
+  then return $ NodeRuntime node sprocket stdIn nodeStdoutFile nodeStderrFile hProcess
+  else left $ NodeExecutableError stdErrContents
 
-  -- The node process can fail on startup, e.g while
-  -- parsing the configuration yaml file. If we don't fail
-  -- when stderr is populated, we end up having to wait for the
-  -- timeout to expire before we can see the error.
   -- TODO: This is flakey. Hydra error:
   -- Unable to run finally: Failure Nothing "\9473\9473\9473 Exception (IOException) \9473\9473\9473\nlsof: readCreateProcess: posix_spawnp: illegal operation (Inappropriate ioctl for device)\n" Nothing
```

Comment:
@newhoggy what was the intention of the call to `lsof`?

Reply:
It checks if the port is open.  We shouldn't continue the test if the port hasn't opened or we may fail to connect.